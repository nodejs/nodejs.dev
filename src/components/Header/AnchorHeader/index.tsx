import React from 'react';
import { AnchorHeaderProps } from '../utils/types'

export default function AnchorHeader({ href, classes, title }: AnchorHeaderProps): JSX.Element {
  return (
    <li className="nav__tabs">
      <a
        className={classes}
        target="_blank"
        href={href}
        rel="noopener noreferrer"
      >
        { title }
      </a>
    </li>
  );
};
