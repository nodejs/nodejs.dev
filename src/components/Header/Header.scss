.activeStyleTab {
  font-weight: 'var(--font-weight-semibold)';
  color: 'var(--color-text-accent)';
  border-bottom: 'var(--space-04) inset var(--color-text-accent)';
}
