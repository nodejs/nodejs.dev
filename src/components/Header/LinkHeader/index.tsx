import React from 'react';
import { Link } from 'gatsby';
import { LinkHeaderProps } from '../utils/types'

export default function LinkHeader({ to, classes, title, activeClassName, partiallyActive }: LinkHeaderProps): JSX.Element {
  return (
    <li className="nav__tabs">
      <Link
        to={to}
        className={classes}
        activeClassName={activeClassName}
        partiallyActive={partiallyActive}
      >
        {title}
      </Link>
    </li>
  );
};
