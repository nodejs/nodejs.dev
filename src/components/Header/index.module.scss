%navItemLi {
  align-items: center;
  cursor: pointer;
  display: flex;
  margin-left: var(--space-16);

  @media (max-width: 800px) {
    display: flex;
    height: 42px;
    margin-left: 0;

    &:not(:last-child) {
      margin-right: 0.5rem;
    }
  }

  > a {
    line-height: 0;
  }

  &:last-child {
    margin-right: var(--space-08);
  }
}

.header {
  background: var(--color-fill-app);
  border-bottom: var(--space-01) solid var(--black2);
  position: sticky;
  top: 0;
  width: 100%;
  z-index: 999;

  .container {
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 0 var(--space-08) 0 var(--space-24);

    @media (max-width: 800px) {
      align-items: start;
      flex-flow: wrap;
      min-height: calc(1 * var(--nav-height));
      padding: 0 var(--space-12);
    }
  }

  .startWrapper {
    align-items: center;
    display: flex;
    justify-content: flex-start;
  }

  .endWrapper {
    align-items: center;
    display: flex;
    flex-grow: 1;
  }

  .logo {
    @media (max-width: 800px) {
      align-items: center;
      display: flex;
      height: var(--nav-height);
    }

    svg {
      height: var(--space-32);

      @media (max-width: 800px) {
        height: 2.4rem;
      }
    }
  }

  .tabs {
    align-items: center;
    display: flex;
    justify-self: flex-start;
    list-style: none;
    margin: 0 0 0 120px;
    margin-right: auto;

    @media (max-width: 1050px) {
      margin: 0;
    }

    @media (max-width: 800px) {
      background: var(--color-fill-app);
      height: var(--nav-height);
      justify-content: center;
      order: 3;
      padding: 0;
      width: 100%;
    }

    > li {
      @extend %navItemLi;

      > a {
        color: var(--color-text-primary);
        padding: calc(var(--space-32) + var(--space-04)) var(--space-16);
        text-decoration: none;

        @media (max-width: 800px) {
          padding: var(--space-08);
        }

        &.active {
          border-bottom: var(--space-04) inset var(--color-text-accent);
          color: var(--color-text-accent);
          font-weight: var(--font-weight-semibold);
          margin-bottom: -4px;
        }

        &:hover {
          border-bottom: 4px solid var(--color-text-primary);
          margin-bottom: -4px;
        }
      }
    }
  }

  .right-container {
    align-items: center;
    display: flex;
    flex: 1 1 auto;
    justify-content: flex-end;
    list-style: none;
    margin: 0;
    padding-left: 0;

    @media (max-width: 800px) {
      padding: 0;
      width: 100%;
    }

    > li {
      @extend %navItemLi;

      &.search-bar {
        align-items: flex-start;
        flex-grow: 1;
        height: 40px;
        justify-content: flex-end;
        position: relative;

        @media (max-width: 1100px) {
          display: none;
        }
      }
    }
  }
}

:global(.dark) {
  .header {
    border-bottom: 1px solid transparent;
  }
}
