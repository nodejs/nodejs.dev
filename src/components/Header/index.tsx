import React from 'react';
import { FormattedMessage } from 'react-intl';
import { LocalizedLink as Link } from 'gatsby-theme-i18n';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGithub } from '@fortawesome/free-brands-svg-icons';
import { ReactComponent as LogoLight } from '../../images/logos/nodejs-logo-light-mode.svg';
import { ReactComponent as LogoDark } from '../../images/logos/nodejs-logo-dark-mode.svg';
import SearchBar from '../SearchBar';
import DarkModeToggle from '../DarkModeToggle';
import LanguageSelector from '../LanguageSelector';
import styles from './index.module.scss';

const Header = (): JSX.Element => (
  <nav aria-label="Primary" className={styles.header}>
    <div className={styles.container}>
      <div className={styles.startWrapper}>
        <Link to="/" aria-label="Homepage">
          <div className={styles.logo}>
            <LogoLight className="light-mode-only" />
            <LogoDark className="dark-mode-only" />
          </div>
        </Link>
      </div>

      <ul className={styles.tabs}>
        <li>
          <Link to="/learn/" activeClassName={styles.active} partiallyActive>
            <FormattedMessage id="components.header.links.learn" />
          </Link>
        </li>
        <li>
          <Link to="/about/" activeClassName={styles.active} partiallyActive>
            <FormattedMessage id="components.header.links.about" />
          </Link>
        </li>
        <li>
          <Link to="/api/" activeClassName={styles.active} partiallyActive>
            <FormattedMessage id="components.header.links.docs" />
          </Link>
        </li>
        <li>
          <Link to="/download/" activeClassName={styles.active} partiallyActive>
            <FormattedMessage id="components.header.links.download" />
          </Link>
        </li>
        <li>
          <Link to="/blog/" activeClassName={styles.active} partiallyActive>
            <FormattedMessage id="components.header.links.blog" />
          </Link>
        </li>
        <li>
          <a href="https://openjsf.org/certification/">
            <FormattedMessage id="components.header.links.certification" />
          </a>
        </li>
      </ul>

      <div className={styles.endWrapper}>
        <ul className={styles.rightContainer}>
          <li className={styles.searchBar}>
            <span className="sr-only">Search Bar</span>
            <SearchBar />
          </li>

          <li>
            <DarkModeToggle />
          </li>

          <li>
            <LanguageSelector />
          </li>

          <li>
            <a
              target="_blank"
              href="https://github.com/nodejs/nodejs.dev"
              rel="noopener noreferrer"
            >
              <span className="sr-only">GitHub</span>
              <FontAwesomeIcon
                icon={faGithub}
                color="var(--color-text-accent)"
                style={{ padding: '1rem', width: '2rem', height: '2rem' }}
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
);

export default Header;
