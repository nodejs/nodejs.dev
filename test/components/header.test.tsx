import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../../src/components/header';

describe('Tests for Header component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
