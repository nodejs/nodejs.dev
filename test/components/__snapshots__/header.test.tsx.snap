// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Header component renders correctly 1`] = `
<nav
  className="nav"
>
  <ul
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "listStyle": "none",
        "margin": "0 auto",
        "padding": "0 4.8rem",
      }
    }
  >
    <li>
      <a
        href="/"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <img
          alt="Node.js"
          className="nav__logo"
          src="test-file-stub"
        />
      </a>
    </li>
    <li
      className="nav__tabs"
    >
      <a
        href="https://nodejs.org/en/docs/"
        target="_blank"
      >
        API Docs
      </a>
    </li>
    <li
      className="nav__tabs"
    >
      <a
        href="https://github.com/nodejs/nodejs.dev"
        target="_blank"
      >
        GitHub
      </a>
    </li>
  </ul>
</nav>
`;
